src/gdb_hostio.c

```diff
@@ -80,7 +82,7 @@ static int hostio_get_response(target_controller_s *const tc)
 /* Interface to host system calls */
 int hostio_open(target_controller_s *tc, target_addr_t path, size_t path_len, target_open_flags_e flags, mode_t mode)
 {
-	gdb_putpacket_f("Fopen,%08X/%X,%08X,%08X", path, path_len, flags, mode);
+	gdb_putpacket_f("Fopen,%08" PRIx32 "/%zX,%08X,%08" PRIx32, path, path_len, flags, mode);
```

Comment:
If these %z's are in the firmware, this will break things. Needs to use `PRIX32` and a cast of the number to uint32_t - eg, `(uint32_t)path_len`

Note that PRIx32 makes the results lower-case hex, while PRIX32 upper-case. We should maintain the case the original formatting string aimed for.

Reply:
I am actually chasing a semihosting hostio bug in the firmware. I should probably post my findings in a separate issue.
For File-I/O F-packets I'm not sure which width should the arguments be in -- target's or gdb host's? (probably not gdb server's which is BMF here) If size_t represents 64-bit ints here, then I reckon PRIX64 will be safer.

Reply:
newlib-nano does not understand 64-bit format specifiers, it's one of the limitations it has.. that's why we're suggesting going to a 32-bit number unless you want to go through the pain we did for the PIDR in adiv5.c.

`path` is an address on the target, `path_len` is the length of that buffer (for all currently supported targets, that limits it to 32-bit, but obviously that changes once we introduce proper 64-bit target support), and flags + mode are the standard integral `open()` values, so also 32-bit there.

That means that all parameters here are described in terms of the target's address space.

Reply:
Understood about the casts, even though I don't expect Aarch32 applications to use paths longer that 65535 symbols even. Amended the commit.
After reordering the hostio-hiding commit behind this, I saw a different solution of keeping `%08X` and introducing `%08lX`, which does not litter the format-strings with `PRIX32` all over the place, hurting readability (including reported compiler warning messages). Should I attempt this?
Will 64-bit targets require 64-bit args and `%016lX` in BMD to support hostio for them, and does mainline GDB support it?

Reply:
No, we are not guaranteed in any way what `uint32_t` maps to - so if we build this with, say, a RISC-V compiler for a potential future version of BMP, we would be sunk again. The `PRI*` macros exist to solve this problem, so please use that solution.

The hope is that we'll be able to rip the format strings out entirely in a future PR where we replace them with direct number formatting instead - eg:

```c
gdb_putpacket_str_part("Fopen,");
gdb_putpacket_upper_hex(path, 8);
gdb_putpacket_char_part('/');
gdb_putpacket_upper_hex(path_len, 8);
gdb_putpacket_char_part(',');
gdb_putpacket_upper_hex(flags, 8);
gdb_putpacket_char_part(',');
gdb_putpacket_upper_hex(mode, 8);
gdb_putpacket_end();
```

If we can get that right we may well be able to save a significant amount of Flash space and all the format string headaches - but this is a ways off for now.

Reply:
Okay. Just saying Farpatch port already includes esp32c3 and allows using BMP on RV32IMC. Not sure about any 64-bit adapters.
The idea with ditching printf and replacing it with fputs/utoa/ftoa is nice, I'd like to follow progess on that (for F4 BMPBootloader reasons).

Reply:
We aren't referring to being able to use BMD with an ESP32-C3 or other RISC-V targets, but rather *running on RISC-V*. Farpatch actually provides another counter-example to switching to `%08lX` - Xtensa architecture ESP32s also have a different definition of `uint32_t` to ARM.

Wasn't going to go with `fputs()` and those other functions as such, but rather using a direct formatted approach like we use in substrate::console_t which winds up producing far better code generation. But yes, that sort of thing.